R/demographics.R

```diff
@@ -87,16 +132,37 @@ montagu_demographic_data <- function(type_code, touchstone_id,
       d <- montagu_demographics_list(touchstone_id, location = location)
       i <- d$id == type_code
       if (sum(i) == 0L) {
-        stop("Could not determine source_code")
+        stop(sprintf("Unknown demographic-statistic-type with id '%s'", type_code))
       } else if (sum(i) > 1L) {
         stop("More than one source_code applicable for this source: ",
              paste(squote(d$source[i]), collapse = ", "))
       }
       source_code <- d$source[i]
     }
-    format <- if (wide) "wide" else NULL
+
+    # Check gender_code is valid
+
+    if (!is.null(gender_code)) {
+      assert_character(gender_code)
+      if (!gender_code %in% c("male", "female", "both")) {
+        stop(sprintf("Invalid gender code '%s' - use male, female or both",
+                     gender_code))
+      }
+
+      if (gender_code %in% c("male", "female")) {
+        d <- montagu_demographics_list(touchstone_id, location)
+        d <- d[d$id == type_code,]
+        if (!d$gendered) {
+          stop(sprintf("Type '%s' is non-gendered and cannot be filtered by '%s'",
+                       type_code, gender_code))
+
+        }
+      }
+    }
+
+    format <- if (wide) "wide" else "long"
     dat <- montagu_demographics_download(touchstone_id, source_code, type_code,
-                                         gender_code, format)
+                                         gender_code, format, NULL, location)
```

Comment:
in the line above write

```
dest <- NULL
```

or use name args here

Reply:
Done